src/Data/Food/Product.elm

```diff
@@ -0,0 +1,618 @@
+
+
+stringToProcessName : String -> ProcessName
+stringToProcessName str =
+    ProcessName str
+
+
+processNameToString : ProcessName -> String
+processNameToString (ProcessName name) =
+    name
+
+
+isProcessing : ProcessName -> Bool
+isProcessing (ProcessName processName) =
+    String.startsWith "Cooking, " processName
+        || String.startsWith "Canning " processName
+        || String.startsWith "Mixing, " processName
+        || String.startsWith "Peeling, " processName
+        || String.startsWith "Fish filleting, " processName
+        || String.startsWith "Slaughtering" processName
+
+
+isWaste : ProcessName -> Bool
+isWaste (ProcessName processName) =
+    String.startsWith "Biowaste " processName
+
+
+isTransport : ProcessName -> Bool
+isTransport (ProcessName processName) =
+    String.startsWith "Transport, " processName
+
+
+isIngredient : ProcessName -> Bool
+isIngredient processName =
+    (isProcessing processName
+        || isWaste processName
+        || isTransport processName
+    )
+        |> not
+
+
+type alias Process =
+    { amount : Amount
+    , impacts : Impacts
+    }
+
+
+type alias ImpactsForProcesses =
+    AnyDict String ProcessName Impacts
+
+
+emptyImpactsForProcesses : ImpactsForProcesses
+emptyImpactsForProcesses =
+    AnyDict.empty processNameToString
+
+
+type alias Processes =
+    AnyDict String ProcessName Process
+
+
+emptyProcesses : Processes
+emptyProcesses =
+    AnyDict.empty processNameToString
+
+
+insertProcess : ProcessName -> Process -> Processes -> Processes
+insertProcess name process processes =
+    AnyDict.insert name process processes
+
+
+computeProcessPefImpact : List Definition -> Processes -> Processes
+computeProcessPefImpact definitions processes =
+    processes
+        |> AnyDict.map
+            (\_ process ->
+                { process
+                    | impacts =
+                        Impact.updatePefImpact definitions process.impacts
+                }
+            )
+
+
+findImpactsByName : ProcessName -> ImpactsForProcesses -> Result String Impacts
+findImpactsByName ((ProcessName name) as procName) =
+    AnyDict.get procName
+        >> Result.fromMaybe ("Procédé introuvable par nom : " ++ name)
+
+
+decodeProcesses : List Definition -> Decoder ImpactsForProcesses
+decodeProcesses definitions =
+    AnyDict.decode (\str _ -> ProcessName str) processNameToString (Impact.decodeImpacts definitions)
+
+
+
+---- Step
+
+
+type alias Step =
+    { ingredients : Processes
+    , transport : Processes
+    , waste : Processes
+    , processing : Processes
+    }
+
+
+type alias Product =
+    { consumer : Step
+    , supermarket : Step
+    , distribution : Step
+    , packaging : Step
+    , plant : Step
+    }
+
+
+type ProductName
+    = ProductName String
+
+
+productNameToString : ProductName -> String
+productNameToString (ProductName name) =
+    name
+
+
+stringToProductName : String -> ProductName
+stringToProductName str =
+    ProductName str
```

Comment:
we don't really NEED it, but I believe it's standard procedure in case of an opaque type: you keep the constructor private to the module, and only export constructors, allowing you to change the implementation at a later stage if needed.

I'd happily remove it if you don't deem it necessary.

Reply:
No no it's a nit so you can keep it if you find it useful!